src/sql/workbench/services/connection/browser/connectionDialogWidget.ts

```diff
@@ -401,6 +413,7 @@ export class ConnectionDialogWidget extends Modal {
 		});
 		this._recentConnectionTree = TreeCreationUtils.createConnectionTree(treeContainer, this.instantiationService, this._configurationService, localize('connectionDialog.recentConnections', "Recent Connections"), controller);
 		if (this._recentConnectionTree instanceof AsyncServerTree) {
+			this._connectionSource = 'recent';
```

Comment:
You could add this at the start of the function because the event will always be `recent` instead of adding here and in the `leftClick` event

Reply:
I actually made a mistake, I should only add this in the event handler.